test/test_api/test_project_locks.py

```diff
@@ -0,0 +1,197 @@
+        err = json.loads(res.data)
+        assert res.status_code == 401, err
+        assert err['status'] == 'failed', err
+        assert err['target'] == 'project', err
+        assert err['exception_cls'] == 'Unauthorized', err
+        assert err['action'] == 'GET', err
+
+    @with_context
+    def test_project_locks_user_subadmin(self):
+        """ Test API should work if user is subadmin"""
+        admin = UserFactory.create(admin=True)
+        subadmin = UserFactory.create(admin=False, subadmin=True)
+
+        project = self.setupProjects()
+        project_id = str(project.id)
+
+        res = self.app.get('/api/locks?id=' + project_id + '&api_key=' + subadmin.api_key + '&all=1')
+        data = json.loads(res.data)
+        assert res.status_code == 200, data
+        assert data[0]['short_name'] == 'test-app1', data
+
+    @with_context
+    def test_project_locks_user_admin(self):
+        """ Test API should work if user is admin"""
+        admin = UserFactory.create(admin=True)
+
+        project = self.setupProjects()
+        project_id = str(project.id)
+
+        res = self.app.get('/api/locks?id=' + project_id + '&api_key=' + admin.api_key + '&all=1')
+        data = json.loads(res.data)
+        assert res.status_code == 200, data
+        assert data[0]['short_name'] == 'test-app1', data
+
+    @with_context
+    def test_project_locks_get_by_id(self):
+        """ Test get locks by project id when project id exists"""
+        admin = UserFactory.create(admin=True)
+        project1 = self.setupProjects()
+
+        # Test get by id
+        res = self.app.get('/api/locks?id=' + str(project1.id) + '&api_key=' + admin.api_key + '&all=1')
+        data = json.loads(res.data)
+        assert res.status_code == 200, data
+        assert len(data) == 1, data
+        assert data[0]['short_name'] == 'test-app1', data
+        assert data[0]['locks'] == [], data
+
+    @with_context
+    def test_project_locks_no_params(self):
+        """ Test API project query when no search params"""
+        admin = UserFactory.create(admin=True)
+        project1 = self.setupProjects()
+
+        # Test no params
+        res = self.app.get('/api/locks?api_key=' + admin.api_key)
+        data = json.loads(res.data)
+        assert res.status_code == 200, data
+        assert len(data) == 0, data
+
+    @with_context
+    def test_project_locks_value_does_not_match(self):
+        """ Test API project query when search value does not match"""
+        admin = UserFactory.create(admin=True)
+        project1 = self.setupProjects()
+
+        # Test value DNE
+        res = self.app.get('/api/locks?id=' + '9999' + '&api_key=' + admin.api_key + '&all=1')
+        data = json.loads(res.data)
+        assert res.status_code == 200, data
+        assert len(data) == 0, data
+
+    @with_context
+    def test_project_locks_param_does_not_exist(self):
+        """ Test API project query when search value does not match"""
```

Comment:
Should this description be `Test API project query when search param does not match`?